src/Repository/CreditRepository.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace App\Repository;
+
+use Wikimedia\ToolforgeBundle\Service\ReplicasClient;
+
+class CreditRepository {
+
+	public function getDatabaseName( $domain, $family ) {
+		if ( $family === 'wikisource' ) {
+			if ( in_array( $domain, [ 'old', 'www', 'mul' ] ) ) {
+				$dbname = 'sourceswiki_p';
+			} elseif ( $domain === 'zh-min-nan' ) {
+				$dbname = 'zh_min_nanwikisource_p';
+			} else {
+				$dbname = $domain . $family . '_p';
+			}
+		} else {
+			$dbname = $domain . $family . '_p';
+		}
+		return $dbname;
+	}
+	
+	public function getPageCredits( ReplicasClient $replicasClient, $lang, $namespaces, $pages ) : array {
+		$dbname = $this->getDatabaseName( $lang, 'wikisource' );
+
+		$conn = $replicasClient->getConnection( $dbname );
+		$pageNames = [];
+		$namespaceName = "";
+		$splittedPages = [];
+	
+		foreach ( $pages as $page ) {
+			$splitPage = explode( ":", $page, 2 );
```

Comment:
What happens if I have a page called `Foo:Baz`? It looks like it will assume `Foo` is a valid namespace when it is not. The code below should somehow account for this, I think.

Reply:
Good point! Done.